tests: smp: cancel CONFIG_MP_NUM_CPUS limit of some testcases

enjiamai opened the pull request:
This PR includes 2 commits:
The first commit try to remove CONFIG_MP_NUM_CPUS=1 configuration for the test of condvar,
sysmutex and semaphore, in order to test SMP condition more.

The second commit makes some changes on codvar test cases to fit SMP.
1. Re-init the mutex to prevent affection from previous test cases.
2. Shorter the test case execution time.
Fixes #33558.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

dcpleung commented on tests/kernel/semaphore/semaphore/src/main.c:
```diff
@@ -1327,11 +1327,11 @@ void test_main(void)
-			 ztest_user_unit_test(test_sem_take_timeout),
-			 ztest_1cpu_user_unit_test(test_sem_take_timeout_forever),
+			 ztest_user_unit_test(test_sem_take_timeout_fails),
+			 ztest_1cpu_user_unit_test(test_sem_take_timeout),
```
Why are you changing the `test_sem_take_timeout` to 1cpu?

enjiamai replied:
Hi, because based on the design of this test case "test_sem_take_timeout", 1cpu is necessary, I believe this not design for multi-core in the first place. It will fail in more than two core. The test logic here is to spawn a thread,  check its count,  then take a semaphore(the spawned thread is scheduled to run). But with multi-core, it would be scheduled right away before the check count, then the test case will fail.
So I make it 1cpu here. Of course, I can slightly change the test code to adapt to multi-core, but I suggest just keep it this way because the test goal is simple here. How do you think about it? Thanks for you review ^^

enjiamai replied:
Hi @dcpleung, do you think should I not make it 1cpu here? thanks~

dcpleung replied:
It has been working without 1cpu. So I think we should keep it that way, especially what you are trying to do in this PR is to enable testing on multiple CPUs.

enjiamai replied:
OK, got it, I think you are right. This makes it more consistent with this PR. I will update this, thanks for your suggestion!

andyross commented on tests/kernel/condvar/condvar_api/src/main.c:
```diff
@@ -557,6 +557,10 @@ void _condvar_usecase(long multi)
+/* Skip these two board when trying to use CONFIG_MP_NUM_CPUS > 1
+ * please see issue #33558
+ */
+#if !defined(CONFIG_BOARD_QEMU_CORTEX_A53) && !defined(CONFIG_X86_64)
```
It's cleaner to write this inline instead of #if'ing the whole test function, e.g.:

```c
 void test_condvar_usecase_signal(void)
{
    if (IS_ENABLED(CONFIG_BOARD_QEMU_CORTEX_A53)
        || IS_ENABLED(CONFIG_X86_64)) {
        /* See #33558  */
        ztest_test_skip();
    }
    _condvar_usecase(0);
}
```

enjiamai replied:
Got it, I have refined it based on your suggestion, thanks for pointing out this!

enjiamai replied:
got learned again, thanks!

andyross commented on tests/kernel/semaphore/semaphore/src/main.c:
```diff
@@ -21,6 +21,7 @@
 #define TOTAL_THREADS_WAITING (5)
 
 #define SEC2MS(s) ((s) * 1000)
+#define QSEC2MS(s) ((s) * 250)
```
Why is this changing units?  Seems like this needs an explanation.

enjiamai replied:
Hi, @andyross , I change this because I got the following message when I run this case.
```
===================================================================
START - test_sem_multiple_take_and_timeouts

    Assertion failed at WEST_TOPDIR/zephyr/subsys/testsuite/ztest/src/ztest.c:125: cpu_hold: (dt < 3000 is false)
1cpu test took too long (5010 ms)
 FAIL - test_sem_multiple_take_and_timeouts in 5.7 seconds
===================================================================
```
So I try to make the time unit smaller to avoid this and save some execution time (without affecting the test case result)

```
===================================================================
START - test_sem_multiple_take_and_timeouts
 PASS - test_sem_multiple_take_and_timeouts in 1.255 seconds
===================================================================

```
Thank you!

Commit:
tests: smp: cancel CONFIG_MP_NUM_CPUS limit of some testcases

Try to remove CONFIG_MP_NUM_CPUS=1 configuration for the test of
condvar, sysmutex and semaphore, in order to test SMP condition more.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

Commit:
tests: kernel: fix two test cases of condvar hang up in SMP

Make some change on two codvar test cases to fit testing under SMP,
and shorter the test cases execution time.

Fixes #33558.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>
